homeassistant/components/neato/vacuum.py

```diff
@@ -198,10 +217,13 @@ def update(self):
 
         self._battery_level = self._state["details"]["charge"]
 
-        if not self._mapdata.get(self._robot_serial, {}).get("maps", []):
+        if self._mapdata is None or not self._mapdata.get(self._robot_serial, {}).get(
+            "maps", []
+        ):
             return
 
-        mapdata = self._mapdata[self._robot_serial]["maps"][0]
+        if self._mapdata:
```

Comment:
Haven't we already checked this above?

Reply:
Yup, added the check above and forgot to remove that one